metadata/register.py

```diff
@@ -58,6 +68,122 @@
                'complex_': PixelsTypecomplex,
                'complex64': PixelsTypecomplex}
 
+
+def get_omexml_bytes(store):
+    # get() is async. Need to sync to get the bytes
+    rsp = store.get("OME/METADATA.ome.xml", prototype=default_buffer_prototype())
+    result = sync(rsp)
+    if result is None:
+        return None
+    return result.to_bytes()
+
+
+def create_fileset():
+    """Create a new Fileset with single OME XML file."""
+    fileset = omero.model.FilesetI()
+    entry = omero.model.FilesetEntryI()
+    # NB: If the clientPath includes .zarr, Bio-Formats tries to import zarr group
+    entry.setClientPath(rstring("OME/METADATA.ome.xml"))
+    fileset.addFilesetEntry(entry)
+
+    # Fill version info
+    system, node, release, version, machine, processor = platform.uname()
+
+    client_version_info = [
+        NamedValue('omero.version', omero_version),
+        NamedValue('os.name', system),
+        NamedValue('os.version', release),
+        NamedValue('os.architecture', machine)
+    ]
+    try:
+        client_version_info.append(
+            NamedValue('locale', locale.getlocale()[0]))
+    except:
+        pass
+
+    upload = omero.model.UploadJobI()
+    upload.setVersionInfo(client_version_info)
+    fileset.linkJob(upload)
+    return fileset
+
+
+def create_settings():
+    """Create ImportSettings and set some values."""
+    settings = omero.grid.ImportSettings()
+    settings.doThumbnails = rbool(True)
```

Comment:
I think that should be `False`. Thumbnails have to be generated afterwards anyway, after the ExternalInfo was set, because the Rendering/Thumbnail service won't have access to the pixel data before that, right?

Reply:
Tried setting to `False` and it still works. So I think it doesn't matter, using `True` just doesn't do anything imo.